tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_subvolgroup_scenarios.py

```diff
@@ -0,0 +1,273 @@
+    """
+    CephFS Subvolumegroup Test
+
+    This test suite validates the functionality of the `ceph fs subvolumegroup` command,
+    covering various scenarios including:
+
+    1. Creating a subvolumegroup with invalid and valid pool layout flags
+    2. Creating a subvolumegroup with desired and invalid modes
+    3. Creating a subvolumegroup with desired and invalid UID/GID
+    4. Creating and resizing a subvolumegroup with specific size
+    5. Creating a subvolumegroup with various options (mode, uid, gid, size)
+
+    """
+    try:
+        tc = "CEPH-83604079"
+        log.info(f"Running CephFS tests for - {tc}")
+        fs_util = FsUtils(ceph_cluster)
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        vol_name = f"cephfs_{random.randint(0, 1000)}"
+        fs_util.create_fs(client1, vol_name)
+        mount_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(5))
+        )
+        fuse_mounting_dir = f"/mnt/cephfs_fuse_{mount_dir}/"
+        fs_util.fuse_mount(
+            [client1], fuse_mounting_dir, extra_params=f" --client_fs {vol_name}"
+        )
+        random_string = "".join(random.choices("abcdefghijklmnopqrstuvwxyz", k=5))
+        # Create a subvolumegroup with invalid pool layout
+        group_name = f"invalid_layout_group_{random_string}"
+        command = f"ceph fs subvolumegroup create {vol_name} {group_name} --pool_layout invalid"
+        log.info(f"Executing command: {command}")
+        out, err = client1.exec_command(sudo=True, cmd=command, check_ec=False)
+        if "Error EINVAL" not in err:
+            log.error(
+                f"Subvolumegroup creation with invalid pool layout should have failed: {out}"
+            )
+            return 1
+        log.info("Subvolumegroup creation with invalid pool layout failed as expected")
+        # Create a subvolumegroup with valid pool layout
+        pool_name = f"cephfs.data.{vol_name}"
+        client1.exec_command(
+            sudo=True, cmd=f"ceph osd pool create {pool_name}  128 128"
+        )
+        client1.exec_command(
+            sudo=True, cmd=f"ceph fs add_data_pool {vol_name} {pool_name}"
+        )
+        group_name = f"valid_layout_group_{random_string}"
+        command = f"ceph fs subvolumegroup create {vol_name} {group_name} --pool_layout {pool_name}"
+        log.info(f"Executing command: {command}")
+        out, err = client1.exec_command(sudo=True, cmd=command)
+        if "Error EINVAL" in err:
+            log.error(f"Subvolumegroup creation with valid pool layout failed: {out}")
+            return 1
+        log.info("Subvolumegroup creation with valid pool layout successful")
+        # Create a subvolumegroup with desired mode
```

Comment:
Can this be validated using subvolumegroup info ?

Reply:
added it